components/rhs_comment/index.js

```diff
@@ -2,14 +2,38 @@
 // See LICENSE.txt for license information.
 
 import {connect} from 'react-redux';
+import {Posts} from 'mattermost-redux/constants';
 import {isChannelReadOnlyById} from 'mattermost-redux/selectors/entities/channels';
 import {getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
 import {getConfig} from 'mattermost-redux/selectors/entities/general';
+import {getPost} from 'mattermost-redux/selectors/entities/posts';
+import {isSystemMessage} from 'mattermost-redux/utils/post_utils';
 
 import {isEmbedVisible} from 'selectors/posts';
 
 import RhsComment from './rhs_comment.jsx';
 
+function isConsecutivePost(state, ownProps) {
+    const post = ownProps.post;
+    const previousPost = ownProps.previousPostId && getPost(state, ownProps.previousPostId);
+
+    let consecutivePost = false;
+
+    if (previousPost) {
+        const postFromWebhook = Boolean(post.props && post.props.from_webhook);
+        const prevPostFromWebhook = Boolean(previousPost.props && previousPost.props.from_webhook);
+        if (previousPost && previousPost.user_id === post.user_id &&
+            post.create_at - previousPost.create_at <= Posts.POST_COLLAPSE_TIMEOUT &&
+            !postFromWebhook && !prevPostFromWebhook &&
+            !isSystemMessage(post) && !isSystemMessage(previousPost) &&
+            (previousPost.root_id === post.root_id || previousPost.id === post.root_id)) {
```

Comment:
We could skip this check since it's the RHS so they're always part of the same thread, but I can see not wanting to do that to keep this code consistent with other places

Reply:
Also , maybe we could check this in the component itself instead of in the `mapStateToProps`? Passing the `previousPost` as a prop would mean the component might re-render a bit more frequently (ie whenever the previous post changes), but it would cut down on the work done in `mapStateToProps`

Reply:
This has proven to work very well on the mobile app, so why not keep it here too... actually what we could do is have this as an utility function in redux